contentful/client.py

```diff
@@ -370,14 +383,54 @@ def _cache_content_types(self):
 
         ContentTypeCache.update_cache(self)
 
+    def _contentful_user_agent(self):
+        """
+        Sets the X-Contentful-User-Agent header.
+        """
+        header = {}
+        from . import __version__
+        header['sdk'] = {
+            'name': 'contentful.py',
+            'version': __version__
+        }
+        if self.application_name is not None:
+            header['app'] = {
+                'name': self.application_name,
+                'version': self.application_version
+            }
+        if self.integration_name is not None:
+            header['integration'] = {
+                'name': self.integration_name,
+                'version': self.integration_version
+            }
+        header['platform'] = {
+            'name': 'python',
+            'version': platform.python_version()
+        }
+        header['os'] = {
+            'name': platform.system(),
+            'version': platform.release()
+        }
+
+        def format_header(key, values):
+            header = "{0} {1}".format(key, values['name'])
+            if values['version'] is not None:
+                header = "{0}/{1}".format(header, values['version'])
+            return header
+
+        result = []
+        for k, values in header.items():
+            result.append(format_header(k, values))
+
+        return '; '.join(result)
+
     def _request_headers(self):
         """
         Sets the default Request Headers.
         """
 
-        from . import __version__
         headers = {
-            'User-Agent': 'PythonContentfulClient/{0}'.format(__version__),
+            'X-Contentful-User-Agent': self._contentful_user_agent(),
```

Comment:
It's just my OCD talking but I think you might have a trailing space here, I'd trim it but that's really just me :D